frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -154,22 +166,46 @@ export class Gallery extends React.Component {
     // app crash telling it cannot render a Link outside a router
     // see https://github.com/tracim/tracim/issues/1637
     // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
-    this.setState({ breadcrumbsList: breadcrumbsList })
+    this.setState({ breadcrumbsList, breadcrumbsLoaded: !includeFile })
   }
 
-  loadFolderDetail = async (workspaceId, folderId) => {
+  loadFolderDetailAndParentsDetails = async (workspaceId, folderId) => {
     const { state, props } = this
 
-    const fetchContentDetail = await handleFetchResult(
+    let folderDetail = {}
+
+    let fetchContentDetail = await handleFetchResult(
       await getFolderDetail(state.config.apiUrl, workspaceId, folderId)
     )
 
     switch (fetchContentDetail.apiResponse.status) {
       case 200:
-        this.setState({ fileName: fetchContentDetail.body.filename })
+        folderDetail.fileName = fetchContentDetail.body.filename
+        folderDetail.folderParentsId = fetchContentDetail.body.parent_id ? [fetchContentDetail.body.parent_id] : []
+
+        let hasReachRootWorkspace = fetchContentDetail.body.parent_id !== null
+        while (hasReachRootWorkspace) {
+          const prevParentId = fetchContentDetail.body.parent_id
+
+          fetchContentDetail = await handleFetchResult(
+            await getFolderDetail(state.config.apiUrl, workspaceId, prevParentId)
+          )
+          if (fetchContentDetail.apiResponse.status === 200) {
+            if (fetchContentDetail.body.parent_id === null || prevParentId === fetchContentDetail.body.parent_id) {
+              hasReachRootWorkspace = false
+            } else {
+              folderDetail.folderParentsId.push(fetchContentDetail.body.parent_id)
+            }
+          } else {
+            this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
+            hasReachRootWorkspace = false
+          }
+        }
         break
-      default: this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
+      default:
+        this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
     }
+    return folderDetail
```

Comment:
if you know `folderDetail` will always have the 2 properties `fileName` and `folderParentsId`, initialize them when you declare your variable